src/common.rs

```diff
@@ -0,0 +1,15 @@
+//! Hal Common Resources - MSP432P401R
+
+pub struct NotDefined;
+pub struct Defined;
+
+unsafe impl Send for NotDefined {}
+
+pub trait Constrain<T> {
+    fn constrain(self) -> T;
+}
+
+pub trait Split<'a> {
```

Comment:
Is it really necessary to keep this lifetime parameter here?

Reply:
I will check it. We need a lifetime parameter to take a reference of the Register by the borrow function  in impl <'a>Split<'a> for TIMER32